api/python/provisioner/commands/validate_setup.py

```diff
@@ -0,0 +1,201 @@
+
+from ..values import UNCHANGED
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class NetworkParamsValidation:
+    cluster_ip: str = NetworkParams.cluster_ip
+    mgmt_vip: str = NetworkParams.mgmt_vip
+    _optional_param = ['cluster_ip', 'mgmt_vip']
+
+    def __attrs_post_init__(self):
+        params = attr.asdict(self)
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class ReleaseParamsValidation:
+    target_build: str = ReleaseParams.target_build
+    _optional_param = []
+
+    def __attrs_post_init__(self):
+        params = attr.asdict(self)
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class StorageEnclosureParamsValidation:
+    type: str = StorageEnclosureParams.type
+    controller_primary_mc_ip: str = (
+                        StorageEnclosureParams.controller_primary_mc_ip)
+    controller_secondary_mc_ip: str = (
+                        StorageEnclosureParams.controller_secondary_mc_ip)
+    controller_user: str = StorageEnclosureParams.controller_user
+    controller_secret: str = StorageEnclosureParams.controller_secret
+    controller_type: str = StorageEnclosureParams.controller_type
+    _optional_param = [
+        'controller_type'
+    ]
+
+    def __attrs_post_init__(self):
+        params = attr.asdict(self)
+        # FIXME why we allow any params for the following types?
+        types = ['JBOD', 'virtual', 'RBOD', 'other']
+        if params['type'] in types:
+            return
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class StorageNodeParamsValidation:
+    hostname: str = StorageEnclosureParams.hostname
```

Comment:
I don't think hostname should come under storage section. can you please verify.

Reply:
@mazinamdar this is just a placeholder, like I had mentioned. 
could be any param, I preferred `hostname` just as a safe one. We can edit later once we have more params to mention in inidividual storage sections, till then I did not want to leave them empty. 
Sample config.ini I used for testing is here : 
https://github.com/Seagate/cortx-prvsnr/wiki/Cluster-Config

Reply:
can it be same as default storage section ? if I want to change controller ip's for only particular node ?